src/lib/swapper/types.ts

```diff
@@ -214,3 +295,44 @@ export type TradeExecutionInput = {
   slippageTolerancePercentageDecimal: string
   getState: () => ReduxState
 }
+
+export type TradeExecutionInput2 = {
+  swapperName: SwapperName
+  tradeQuote: TradeQuote
+  stepIndex: number
+  slippageTolerancePercentageDecimal: string
+  evm: EvmTradeExecutionProps
+  utxo: UtxoTradeExecutionProps
+  cosmosSdk: CosmosSdkTradeExecutionProps
+}
+
+export enum TradeExecutionEvent {
+  SellTxHash = 'sellTxHash',
+  Status = 'status',
+  Success = 'success',
+  Fail = 'fail',
+  Error = 'error',
+}
+
+export type SellTxHashArgs = { stepIndex: number; sellTxHash: string }
+export type StatusArgs = {
+  stepIndex: number
+  status: TxStatus
+  message?: string
+  buyTxHash?: string
+}
+
+export type TradeExecutionEventMap = {
+  [TradeExecutionEvent.SellTxHash]: (args: SellTxHashArgs) => void
+  [TradeExecutionEvent.Status]: (args: StatusArgs) => void
+  [TradeExecutionEvent.Success]: (args: StatusArgs) => void
+  [TradeExecutionEvent.Fail]: (args: StatusArgs) => void
+  [TradeExecutionEvent.Error]: (args: unknown) => void
+}
+
+export interface TradeExecutionBase {
+  on<T extends TradeExecutionEvent>(eventName: T, callback: TradeExecutionEventMap[T]): void
+
+  exec?: (input: TradeExecutionInput) => Promise<{ cancelPolling: () => void } | void>
+  exec2?: (input: TradeExecutionInput2) => Promise<{ cancelPolling: () => void } | void>
```

Comment:
2 methods for convenience, really need to avoid name collisions to prevent casting and total destruction of type safety in the process

Reply:
maybe f2 to execWalletAgnostic? fine with the copy paste just maybe a lil moar descriptive?